static/js/settings_notifications.js

```diff
@@ -9,101 +9,39 @@ var notification_settings = ["enable_stream_desktop_notifications",
     "enable_online_push_notifications","enable_digest_emails"];
 
 exports.set_up = function () {
-    $("#notify-settings-status").hide();
+    var notify_settings_status = $("#notify-settings-status").expectOne();
+    notify_settings_status.hide();
 
     if (!page_params.realm_show_digest_email) {
         $("#other_notifications").hide();
     }
 
-    function update_notification_settings_success(resp, statusText, xhr) {
-        var result = JSON.parse(xhr.responseText);
-        var notify_settings_status = $('#notify-settings-status').expectOne();
-
-        // Stream notification settings.
-
-        _.each(result, function  (v, k) {
-            if (_.has(notification_settings, k) && result[k] !== undefined) {
-                page_params[k] = result[k];
+    _.each(notification_settings, function (setting) {
+         $("#"+setting).change(function () {
+            var data = {};
+            data[setting] = JSON.stringify(this.checked);
+            channel.patch({
+                url: '/json/settings/notifications',
+                data: data,
+                success: function () {
+                    var setting_name = $('label[for=' + setting + ']').text().trim();
+                    if (data[setting] === 'true') {
+                        ui_report.success(i18n.t("Turned on: " + setting_name), notify_settings_status);
```

Comment:
Ahh, what I mean is I think we might want a map from machine-readable names like "enable_desktop_notifications" to i18n.t("desktop notifications") so that we can combine these to say "Enabled desktop notifications" in all languages.